features/twitter.feature

```diff
@@ -10,5 +10,31 @@ Feature: Tweeting Live Events
       | Scrum      | Daily scrum meeting     | Scrum           | 2014/02/03 07:00:00 UTC | 150      | never   | UTC       |         | And an event "Scrum"                      |                       |
     And an event "Scrum"
 
-  Scenario: Event going live causes tweets of hangout link and youtube link to be sent
-    When the HangoutConnection has pinged to indicate the event start, appropriate tweets will be sent
+  Scenario: Event going live without valid live stream does not cause youtube link to be tweeted
+    Given the live stream has not started
+    When the HangoutConnection has pinged to indicate the event start
+    Then the youtube link will not be sent
+
+  Scenario: Event going live without valid live stream will have youtube link tweeted later when live
+    Given the live stream has not started
+    When the HangoutConnection has pinged to indicate the event start
+    Then the youtube link will not be sent
+    And the live stream has started
+    And the HangoutConnection has pinged to indicate the event continuing
+    Then the youtube link will be sent
```

Comment:
This scenario fails because the logic in the controller switches on whether the yt_id has changed or not.  The first ping to indicate a start, the event goes live, but the livestream on youtube has not yet gone live (this always happens).  Then the second ping the livestream has gone live but the youtube id has not changed because it is the same as the last ping.  Therefore the tweet never gets sent.

A quick fix for this might be to use this scenario to drive a change to the model of the event instance to include a youtube_tweet_sent boolean that defaults to false and is set to true the first time the youtube link has been tweeted.

To see where the yt_video_id_changed logic is go here: https://github.com/AgileVentures/WebsiteOne/blob/88dcd516fb84ef4921b203a829e0d1d7aa8cfd9a/app/controllers/event_instances_controller.rb#L10